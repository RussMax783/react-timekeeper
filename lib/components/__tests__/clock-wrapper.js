'use strict';

var _enzyme = require('enzyme');

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactTestRenderer = require('react-test-renderer');

var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);

var _ClockWrapper = require('../ClockWrapper');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

describe('ClockWrapper component', function () {
	var hour = 12;
	var minute = 30;
	var unit = 'hour';

	var changeHour = jest.fn();
	var changeMinute = jest.fn();
	var changeMeridiem = jest.fn();

	test('renders correctly', function () {
		var tree = _reactTestRenderer2.default.create(_react2.default.createElement(_ClockWrapper.ClockWrapper, {
			config: {},
			unit: unit,
			hour: hour,
			minute: minute,
			meridiem: 'pm',
			changeHour: changeHour,
			changeMinute: changeMinute,
			changeMeridiem: changeMeridiem
		})).toJSON();
		expect(tree).toMatchSnapshot();
	});

	test('triggers change meridiem', function () {
		var rendered = (0, _enzyme.shallow)(_react2.default.createElement(_ClockWrapper.ClockWrapper, {
			config: {},
			unit: unit,
			hour: hour,
			minute: minute,
			meridiem: 'pm',
			changeHour: changeHour,
			changeMinute: changeMinute,
			changeMeridiem: changeMeridiem
		}));
		rendered.find('button.type_am').simulate('click');
		expect(changeMeridiem).lastCalledWith('am');

		rendered.find('button.type_pm').simulate('click');
		expect(changeMeridiem).lastCalledWith('pm');
	});
});