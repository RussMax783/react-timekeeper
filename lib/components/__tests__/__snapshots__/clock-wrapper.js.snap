exports[`ClockWrapper component renders correctly 1`] = `
<div
  className="react-timekeeper__clock-wrapper"
  style={
    Object {
      "background": undefined,
      "padding": "18px 0 14px",
      "textAlign": "center",
    }
  }>
  <div
    className="react-timekeeper__clock"
    data-radium={true}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "background": undefined,
        "borderRadius": "200px",
        "cursor": "pointer",
        "display": "inline-block",
        "height": "220px",
        "position": "relative",
        "width": "220px",
      }
    }>
    <div
      className="react-timekeeper__clock-animations-wrapper"
      data-radium={true}>
      <div
        className="react-timekeeper__clock-animations"
        data-radium={true}
        style={
          Object {
            "position": "absolute",
          }
        }>
        <span
          data-radium={true}
          style={
            Object {
              "borderRadius": "99px",
              "color": undefined,
              "display": "inline-block",
              "fontSize": "16px",
              "height": 34,
              "left": 136.99999999999997,
              "lineHeight": "34px",
              "opacity": 1,
              "pointerEvents": "none",
              "position": "absolute",
              "textAlign": "center",
              "top": 16.789764466969388,
              "width": 34,
              "zIndex": 5,
            }
          }>
          1
        </span>
        <span
          data-radium={true}
          style={
            Object {
              "borderRadius": "99px",
              "color": undefined,
              "display": "inline-block",
              "fontSize": "16px",
              "height": 34,
              "left": 169.21023553303058,
              "lineHeight": "34px",
              "opacity": 1,
              "pointerEvents": "none",
              "position": "absolute",
              "textAlign": "center",
              "top": 48.99999999999996,
              "width": 34,
              "zIndex": 5,
            }
          }>
          2
        </span>
        <span
          data-radium={true}
          style={
            Object {
              "borderRadius": "99px",
              "color": undefined,
              "display": "inline-block",
              "fontSize": "16px",
              "height": 34,
              "left": 181,
              "lineHeight": "34px",
              "opacity": 1,
              "pointerEvents": "none",
              "position": "absolute",
              "textAlign": "center",
              "top": 92.99999999999999,
              "width": 34,
              "zIndex": 5,
            }
          }>
          3
        </span>
        <span
          data-radium={true}
          style={
            Object {
              "borderRadius": "99px",
              "color": undefined,
              "display": "inline-block",
              "fontSize": "16px",
              "height": 34,
              "left": 169.21023553303058,
              "lineHeight": "34px",
              "opacity": 1,
              "pointerEvents": "none",
              "position": "absolute",
              "textAlign": "center",
              "top": 137,
              "width": 34,
              "zIndex": 5,
            }
          }>
          4
        </span>
        <span
          data-radium={true}
          style={
            Object {
              "borderRadius": "99px",
              "color": undefined,
              "display": "inline-block",
              "fontSize": "16px",
              "height": 34,
              "left": 137.00000000000006,
              "lineHeight": "34px",
              "opacity": 1,
              "pointerEvents": "none",
              "position": "absolute",
              "textAlign": "center",
              "top": 169.21023553303058,
              "width": 34,
              "zIndex": 5,
            }
          }>
          5
        </span>
        <span
          data-radium={true}
          style={
            Object {
              "borderRadius": "99px",
              "color": undefined,
              "display": "inline-block",
              "fontSize": "16px",
              "height": 34,
              "left": 93.00000000000003,
              "lineHeight": "34px",
              "opacity": 1,
              "pointerEvents": "none",
              "position": "absolute",
              "textAlign": "center",
              "top": 181,
              "width": 34,
              "zIndex": 5,
            }
          }>
          6
        </span>
        <span
          data-radium={true}
          style={
            Object {
              "borderRadius": "99px",
              "color": undefined,
              "display": "inline-block",
              "fontSize": "16px",
              "height": 34,
              "left": 49,
              "lineHeight": "34px",
              "opacity": 1,
              "pointerEvents": "none",
              "position": "absolute",
              "textAlign": "center",
              "top": 169.21023553303058,
              "width": 34,
              "zIndex": 5,
            }
          }>
          7
        </span>
        <span
          data-radium={true}
          style={
            Object {
              "borderRadius": "99px",
              "color": undefined,
              "display": "inline-block",
              "fontSize": "16px",
              "height": 34,
              "left": 16.789764466969416,
              "lineHeight": "34px",
              "opacity": 1,
              "pointerEvents": "none",
              "position": "absolute",
              "textAlign": "center",
              "top": 137.00000000000006,
              "width": 34,
              "zIndex": 5,
            }
          }>
          8
        </span>
        <span
          data-radium={true}
          style={
            Object {
              "borderRadius": "99px",
              "color": undefined,
              "display": "inline-block",
              "fontSize": "16px",
              "height": 34,
              "left": 5,
              "lineHeight": "34px",
              "opacity": 1,
              "pointerEvents": "none",
              "position": "absolute",
              "textAlign": "center",
              "top": 93.00000000000003,
              "width": 34,
              "zIndex": 5,
            }
          }>
          9
        </span>
        <span
          data-radium={true}
          style={
            Object {
              "borderRadius": "99px",
              "color": undefined,
              "display": "inline-block",
              "fontSize": "16px",
              "height": 34,
              "left": 16.78976446696936,
              "lineHeight": "34px",
              "opacity": 1,
              "pointerEvents": "none",
              "position": "absolute",
              "textAlign": "center",
              "top": 49.00000000000007,
              "width": 34,
              "zIndex": 5,
            }
          }>
          10
        </span>
        <span
          data-radium={true}
          style={
            Object {
              "borderRadius": "99px",
              "color": undefined,
              "display": "inline-block",
              "fontSize": "16px",
              "height": 34,
              "left": 49.00000000000003,
              "lineHeight": "34px",
              "opacity": 1,
              "pointerEvents": "none",
              "position": "absolute",
              "textAlign": "center",
              "top": 16.789764466969388,
              "width": 34,
              "zIndex": 5,
            }
          }>
          11
        </span>
        <span
          data-radium={true}
          style={
            Object {
              "borderRadius": "99px",
              "color": undefined,
              "display": "inline-block",
              "fontSize": "16px",
              "height": 34,
              "left": 92.99999999999997,
              "lineHeight": "34px",
              "opacity": 1,
              "pointerEvents": "none",
              "position": "absolute",
              "textAlign": "center",
              "top": 5,
              "width": 34,
              "zIndex": 5,
            }
          }>
          12
        </span>
        <svg
          className="react-timekeeper__clock-svgs"
          data-radium={true}
          height={220}
          style={
            Object {
              "opacity": 1,
              "position": "relative",
            }
          }
          viewBox="0 0 220 220"
          width={220}
          xmlns="http://www.w3.org/2000/svg">
          <g
            data-radium={true}
            transform="rotate(360 110 110)">
            <line
              stroke={undefined}
              strokeWidth="1"
              x1={110}
              x2={110}
              y1={110}
              y2={20} />
            <circle
              cx={110}
              cy={110}
              fill={undefined}
              r={1.5} />
            <circle
              cx={110}
              cy={22}
              fill={undefined}
              r={17} />
          </g>
        </svg>
      </div>
    </div>
  </div>
  <div
    className="react-timekeeper__meridiem-toggle-wrapper"
    style={
      Object {
        "marginTop": "-16px",
        "padding": "0 30px",
        "position": "relative",
        "textAlign": "left",
        "zIndex": 10,
      }
    }>
    <button
      className="react-timekeeper-button-reset react-timekeeper__meridiem-toggle type_am "
      onClick={[Function]}
      style={
        Array [
          Object {
            "background": undefined,
            "borderRadius": "99px",
            "color": undefined,
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": undefined,
            "fontSize": "14px",
            "height": 38,
            "lineHeight": "38px",
            "padding": 0,
            "textAlign": "center",
            "transition": "0.15s ease-out",
            "width": 38,
          },
          false,
        ]
      }
      type="button">
      AM
    </button>
    <button
      className="react-timekeeper-button-reset react-timekeeper__meridiem-toggle type_pm"
      onClick={[Function]}
      style={
        Array [
          Object {
            "background": undefined,
            "borderRadius": "99px",
            "color": undefined,
            "cursor": "pointer",
            "display": "inline-block",
            "fontFamily": undefined,
            "fontSize": "14px",
            "height": 38,
            "lineHeight": "38px",
            "padding": 0,
            "textAlign": "center",
            "transition": "0.15s ease-out",
            "width": 38,
          },
          Object {
            "float": "right",
          },
          Object {
            "background": undefined,
          },
        ]
      }
      type="button">
      PM
    </button>
  </div>
</div>
`;
